plutus-contract/test/Spec/Contract/Tx/Constraints/RequiredSigner.hs

```diff
@@ -156,3 +169,98 @@ mustBeSignedByTypedValidator = Scripts.mkTypedValidator @UnitTest
     $$(PlutusTx.compile [|| wrap ||])
     where
         wrap = Scripts.mkUntypedValidator
+
+
+-- plutus-tx-constraints tests
+-- all below to be covered by the above tests when the corresponding constraints will be implemented
+-- for CardanoTx
+
+
+{-# INLINABLE mustReferenceOutputValidatorV2 #-}
+mustReferenceOutputValidatorV2 :: PV2.Validator
+mustReferenceOutputValidatorV2 = PV2.mkValidatorScript
+    $$(PlutusTx.compile [|| wrap ||])
+ where
+     mkMustReferenceOutputV2Validator = mkMustReferenceOutputValidator Cons.V2.checkScriptContext
+     wrap = PSU.V2.mkUntypedValidator mkMustReferenceOutputV2Validator
+
+tag :: Trace.ContractInstanceTag
+tag = "instance 1"
+
+mkMustReferenceOutputValidator
+    :: (Constraints.TxConstraints Void Void -> sc -> Bool)
+    -> PV1.TxOutRef -> () -> sc -> Bool
+mkMustReferenceOutputValidator checkScriptContext txOutRef _ =
+    checkScriptContext (Constraints.mustReferenceOutput txOutRef)
+
+mustReferenceOutputV2ValidatorAddress :: Ledger.Address
+mustReferenceOutputV2ValidatorAddress =
+    PSU.V2.mkValidatorAddress mustReferenceOutputValidatorV2
+
+cardanoTxOwnWalletContract
+    :: Ledger.PaymentPubKey
+    -> Ledger.PaymentPubKeyHash
+    -> Contract () EmptySchema ContractError ()
+cardanoTxOwnWalletContract pk pkh = do
+    let mkTx lookups constraints = either (error . show) id $ TxCons.mkTx @Any def lookups constraints
+
+    utxos <- ownUtxos
+    myAddr <- ownAddress
+    let get3 (a:b:c:_) = (a, b, c)
+        get3 _         = error "Spec.Contract.TxConstraints.get3: not enough inputs"
+        ((utxoRef, utxo), (utxoRefForBalance1, _), (utxoRefForBalance2, _)) = get3 $ M.toList utxos
+        vh = fromJust $ Ledger.toValidatorHash mustReferenceOutputV2ValidatorAddress
+        lookups1 = Constraints.unspentOutputs utxos
+               <> Constraints.plutusV2OtherScript mustReferenceOutputValidatorV2
+               <> Constraints.paymentPubKey pk
```

Comment:
We really should have a `Constraints.paymentPubKeyHash` function...

Reply:
True. @berewt Shouldn't be too hard to add it in this PR?